cpp/src/io/comp/comp.cpp

```diff
@@ -142,7 +143,8 @@ uint8_t* emit_literal(uint8_t* out_begin, uint8_t const* literal_begin, uint8_t
     // Fits into a single tag byte
     *out_it++ = n << 2;
   } else {
-    auto const log2_n = 31 - __builtin_clz(n);
+    // log2(n) is 32 - 1 - __builtin_clz(n)
+    auto const log2_n = numeric::detail::count_significant_bits(static_cast<uint32_t>(n)) - 1;
```

Comment:
I don't think the comment is necessary, the new code is more direct than what I had before.